src/cascadia/TerminalControl/XamlUiaTextRange.h

```diff
@@ -0,0 +1,75 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- XamlUiaTextRange.h
+
+Abstract:
+- This module is a wrapper for the UiaTextRange
+  (a text range accessibility provider). It allows
+  for UiaTextRange to be used in Windows Terminal.
+- Wraps the UIAutomationCore ITextRangeProvider
+  (https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-itextrangeprovider)
+  with a XAML ITextRangeProvider
+  (https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.provider.itextrangeprovider)
+
+Author(s):
+- Carlos Zamora   (CaZamor)    2019
+--*/
+
+#pragma once
+
+#include "TermControlAutomationPeer.h"
+#include <UIAutomationCore.h>
+#include <winrt/Windows.UI.Xaml.h>
+#include "../types/UiaTextRange.hpp"
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    class XamlUiaTextRange :
+        public winrt::implements<XamlUiaTextRange, Windows::UI::Xaml::Automation::Provider::ITextRangeProvider>
+    {
+    public:
+        XamlUiaTextRange(::ITextRangeProvider* uiaProvider, Windows::UI::Xaml::Automation::Provider::IRawElementProviderSimple parentProvider) :
+            _parentProvider{ parentProvider }
+        {
+            _uiaProvider.attach(uiaProvider);
+        }
+
+        // ITextRangeProvider

```

Comment:
All of these are ITextRangeProvider? Or does it end somewhere below and you didn't note where?